File: tests/src/com.google.protobuf/protobuf-javalite/3.25.5/src/test/java/com_google_protobuf/protobuf_javalite/ExtensionRegistryLiteTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.google.protobuf.Empty;
import com.google.protobuf.Extension;
import com.google.protobuf.ExtensionLite;
import com.google.protobuf.ExtensionRegistryLite;
import com.google.protobuf.WireFormat;
import org.junit.jupiter.api.Test;

public class ExtensionRegistryLiteTest {
    private static final String FULL_REGISTRY_CLASS_NAME = "com.google.protobuf.ExtensionRegistry";
    private static final int STRING_EXTENSION_NUMBER = 12345;

    @Test
    public void addExtensionLiteDelegatesFullRuntimeExtensionsToFullRegistry() {
        ExtensionRegistryLite registry = ExtensionRegistryLite.newInstance();
        ExtensionLite<Empty, String> extension = new NoOpFullRuntimeExtension();

        registry.add(extension);

        assertEquals(FULL_REGISTRY_CLASS_NAME, registry.getClass().getName());
    }

    private static final class NoOpFullRuntimeExtension extends Extension<Empty, String> {
        @Override
        public int getNumber() {
            return STRING_EXTENSION_NUMBER;
        }

        @Override
        public WireFormat.FieldType getLiteType() {
            return WireFormat.FieldType.STRING;
        }

        @Override
        public boolean isRepeated() {
            return false;
        }

        @Override
        public String getDefaultValue() {
            return "";
        }

        @Override
        public Empty getMessageDefaultInstance() {
            return null;
        }
    }
}

File: tests/src/com.typesafe.akka/akka-protobuf-v3_2.13/2.9.0-M1/src/test/scala/com_typesafe_akka/akka_protobuf_v3_2_13/ExtensionRegistryLiteTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_typesafe_akka.akka_protobuf_v3_2_13

import akka.protobufv3.internal.ExtensionLite
import akka.protobufv3.internal.ExtensionRegistry
import akka.protobufv3.internal.ExtensionRegistryLite
import akka.protobufv3.internal.GeneratedMessage
import akka.protobufv3.internal.Message
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test

class ExtensionRegistryLiteTest {
  @Test
  def addFullRuntimeExtensionThroughLiteApiRegistersItInFullRegistry(): Unit = {
    val extension: GeneratedMessage.GeneratedExtension[Message, GeneratedDescriptorDependency] =
      GeneratedMessage.newFileScopedGeneratedExtension(
        classOf[GeneratedDescriptorDependency],
        null,
        classOf[GeneratedDescriptorDependency].getName,
        "covered_extension"
      )
    val registry: ExtensionRegistryLite = ExtensionRegistry.newInstance()

    registry.add(extension: ExtensionLite[Message, GeneratedDescriptorDependency])

    val fullRegistry: ExtensionRegistry = registry.asInstanceOf[ExtensionRegistry]
    val descriptor = extension.getDescriptor
    val infoByName = fullRegistry.findMutableExtensionByName(descriptor.getFullName)
    assertNotNull(infoByName)
    assertSame(descriptor, infoByName.descriptor)
    assertSame(infoByName, fullRegistry.findMutableExtensionByNumber(descriptor.getContainingType, descriptor.getNumber))
  }
}
